api_tests/preprints/views/test_preprint_actions.py

```diff
@@ -6,12 +6,12 @@
 )
 from osf.utils import permissions as osf_permissions
 
-from api_tests.reviews.mixins.filter_mixins import ReviewActionFilterMixin
+from api_tests.users.views.test_user_actions import TestReviewActionFilters
 from api_tests.reviews.mixins.comment_settings import ReviewActionCommentSettingsMixin
 
 
 @pytest.mark.enable_quickfiles_creation
-class TestPreprintActionFilters(ReviewActionFilterMixin):
+class TestPreprintActionFilters(TestReviewActionFilters):
```

Comment:
> but i think a success criterion here is that test_filter_actions (or something like it) should be run on both /v2/actions/reviews/ and /v2/preprints/<id>/review_actions/

that's not mentioned in the ticket. I also just feel using inheritance in tests is wrong, It makes it difficult to identify specific problems. I end up jumping from file to file. It's musch more convenient to have a single set of tests per endpoint.  

The url formulation here is a bit non-standard too,  should everything be `/v2/users/<user_id>/actions/reviews/` then filter by node/preprint. I'm not super familiar with how we do review actions.

Reply:
sorry for distracting you with producty jargon -- by "success criterion" just meant "something that should be true for this to be considered done"

----

before this PR, our tests checked that filtering worked on those two lists of actions.

after this PR, our tests check that filtering works on one of those lists, but not the other.

thing that should generally be true: we don't remove tests unless we mean to.

----

i'm good with rearranging/cleaning tests to use less inheritance, but (1) you're still using inheritance in a confusing way, and (2) be careful while rearranging that you don't create new gaps in our test coverage

Reply:
So my understanding initially changing the name from `ReviewActionFilterMixin` to `TestReviewActionFilters` made the test classes run independently without having to inherit from each other. I moved a lot of methods to make that possible, but I need to look at this more carefully because you're right that these changes probably aren't necessary.

Reply:
yes i see the `Test` prefix, but i don't think that's the problem -- it might be easier to discuss if we lay out the test classes one by one:

- `TestReviewActionFilters` (from `api_tests.reviews.views.test_review_action_filter`)
  - url: `/v2/actions/reviews/`
  - tests: `test_filter_actions`, `test_no_permission`
- `TestReviewActions` (from `api_tests.users.views.test_user_actions`)
  - url: `/v2/actions/reviews/`
  - tests: `test_no_permission`
- `TestPreprintActionFilters ` (from `api_tests.preprints.views.test_preprint_actions`)
  - inherits `TestReviewActions`
  - url: `/v2/preprints/<guid>/review_actions/`
  - tests: `test_unauthorized_user`, `test_no_permission` (overrides `TestReviewActions.test_no_permission`)

the gap i'm talking about is testing that the `filter` query param works (as in `test_filter_actions`) on `/v2/preprints/<guid>/review_actions/`

it sounds like you meant to either:
1. have `TestPreprintActionFilters` inherit `TestReviewActionFilters` (instead of `TestReviewActions`),
2. implement `test_filter_actions` on `TestReviewActions` (instead of `TestReviewActionFilters`), or
3. duplicate `test_filter_actions` as separate tests on `TestReviewActionFilters` and `TestPreprintActionFilters`, while making `TestPreprintActionFilters` no longer inherit another test class

any one of those three would fix the problem, as i see it -- option (3) would be closest to your goal to make "the test classes run independently without having to inherit from each other"